lib/validator/v3/validator_test.go

```diff
@@ -1947,6 +1947,64 @@ func init() {
 				},
 			}, true,
 		),
+		Entry("allow a Service match in an egress rule",
+			&api.NetworkPolicy{
+				ObjectMeta: v1.ObjectMeta{Name: "thing"},
+				Spec: api.NetworkPolicySpec{
+					Egress: []api.Rule{
+						{
+							Action: "Allow",
+							Destination: api.EntityRule{
+								Services: &api.ServiceMatch{
+									Name:      "service1",
+									Namespace: "default",
+								},
+							},
+						},
+					},
+				},
+			}, true,
+		),
+		Entry("disallow a Service match in an ingress rule",
+			&api.NetworkPolicy{
+				ObjectMeta: v1.ObjectMeta{Name: "thing"},
+				Spec: api.NetworkPolicySpec{
+					Ingress: []api.Rule{
+						{
+							Action: "Allow",
+							Source: api.EntityRule{
+								Services: &api.ServiceMatch{
+									Name:      "service1",
+									Namespace: "default",
+								},
+							},
+						},
+					},
+				},
+			}, false,
+		),
+		Entry("disallow a Service match AND a ServiceAccount match",
+			&api.NetworkPolicy{
+				ObjectMeta: v1.ObjectMeta{Name: "thing"},
+				Spec: api.NetworkPolicySpec{
+					Ingress: []api.Rule{
+						{
+							Action: "Allow",
+							Destination: api.EntityRule{
+								ServiceAccounts: &api.ServiceAccountMatch{
+									Names: []string{"serviceaccount"},
+								},
+								Services: &api.ServiceMatch{
+									Name:      "service1",
+									Namespace: "default",
+								},
+							},
+						},
+					},
+				},
+			}, false,
+		),
+
```

Comment:
Worth shooting for full coverage in the validation logic; think there are a few more cases to cover